### ISSUE TITLE
Broken docs links in VS 2015 Extension 

### ISSUE THREAD

**Issue by mmnosek** [Starcounter/Starcounter#3791]
In Visual Studio 2015 "Extension & Updates" window there is a list of installed extensions. When you try to follow links in description of Starcounter VS2015 the 404 page on starcounter.com appears. This issue is about two links:
- More information http://starcounter.com/wiki/MoreInfoAboutExtension14
- Getting started http://starcounter.com/wiki/GettingStartedFromExtension14

**Comment by dan31** [comment]
Looked now. This is still a case + design of the extension is obsolete (old icon). Might be great to package this lifting work together with delivering VS 2017 extension.

**Comment by per-samuelsson** [comment]
> Might be great to package this lifting work together with delivering VS 2017 extension.

Sure. Then related: #4120.

**Comment by per-samuelsson** [comment]
@Mackiovello 

Can you help me out creating some good landing pages for the Visual Studio extension? The content initially is not important (could be blank), just some links I can point to where we could write something eventually.

With the Starcounter extension installed, here's how it look in Visual Studio 2015 currently (`Tools | Extensions and Updates`):

![image](https://cloud.githubusercontent.com/assets/2902045/23741743/587579b2-04aa-11e7-9b54-10556f26dbc2.png)

To the right, its those More information / Getting Started links we want to fix.

After #4120, we'll have two extensions: one for VS 2015 and one for VS 2107. At the moment, they work more or less equally, but maybe it's good to have separate landing pages for each (4 links in total), and then direct them to a common ones as long as they match in functionality. WDYT?

2015|2017
-----|-----
More information|More information
Getting started|Getting started

**Comment by Mackiovello** [comment]
> Can you help me out creating some good landing pages for the Visual Studio extension? The content initially is not important (could be blank), just some links I can point to where we could write something eventually.

What kind of information do we want on this page? Is it a general Starcounter text of some sort?

**Comment by per-samuelsson** [comment]
> What kind of information do we want on this page? Is it a general Starcounter text of some sort?

I'm not sure TBH. Maybe @mmnosek or someone from the app team could advice?

But let's pretend I'm a developer (of apps) and I see that I somehow have this Starcounter extension in Visual Studio and I know nothing about it, where would I want to land when clicking [More information](?)? And [Getting started](?)?

I guess, for the former, on some spot teaching me about what Starcounter is in general (audience: developer), and for the later, some tutorial where I'm shown how to build my first app in Visual Studio?

**Comment by per-samuelsson** [comment]
I guess what would give us the greatest flexibility would be to reserve 4 URIs for the purpose, and postpone where they actually redirect you. Especially considering links are bundled with the software package and not easy to change once the go into the installer.

So something like:

2015 More Information - > starcounter.io/StarcounterVisualStudio2015MoreInfo
2015 Getting Started - > starcounter.io/StarcounterVisualStudio2015GettingStarted
2017 More Information - > starcounter.io/StarcounterVisualStudio2017MoreInfo
2017 Getting Started - > starcounter.io/StarcounterVisualStudio2017GettingStarted

Then we initially just redirect both "More Info" to https://docs.starcounter.io/ and Getting Started to https://docs.starcounter.io/tutorial/.

If that is to ambitious, I can just use those links straight away I guess, and we're done with it. You guys choose 😎

**Comment by per-samuelsson** [comment]
Also, @dan31, regarding:

> ...design of the extension is obsolete (old icon). Might be great to package this lifting work together with delivering VS 2017 extension.

If you want us to use some new icon and preview image, ship them this way please. Or you file an epic @dan31 commit to Level1 yourself! 🥇

For the icon file, this is the source:

https://github.com/Starcounter/Starcounter/blob/develop/src/Development/Starcounter.ico

For the preview image, this one:

https://github.com/Starcounter/Starcounter/blob/develop/src/Development/StarcounterLogo.jpg

My [picture above](https://github.com/Starcounter/Starcounter/issues/3791#issuecomment-285287676) show how they currently render.

**Comment by dan31** [comment]
@andwah could you please supply these locations with the freshest logotypes?
https://github.com/Starcounter/Starcounter/blob/develop/src/Development/Starcounter.ico
https://github.com/Starcounter/Starcounter/blob/develop/src/Development/StarcounterLogo.jpg
Currently they contain old. Thanks in advance!

**Comment by andwah** [comment]
@dan31 Looking into it now.

**Comment by andwah** [comment]
Pushed new icons to `v2.3` and `develop`

**Comment by mmnosek** [comment]
> I'm not sure TBH. Maybe @mmnosek or someone from the app team could advice?

From my perspective, there is no reason to create separate pages. We want neither to distribute information between too many pages nor create separate pages just because we have some links in VS extension. I support current idea by @per-samuelsson:
> Then we initially just redirect both "More Info" to https://docs.starcounter.io/ and Getting Started to https://docs.starcounter.io/tutorial/.

This is perfectly fine! 👍

**Comment by Mackiovello** [comment]
> From my perspective, there is no reason to create separate pages. We want neither to distribute information between too many pages nor create separate pages just because we have some links in VS extension. I support current idea by @per-samuelsson:

I do agree with this. Although, are there Visual Studio specific things that we might want to document? I'm not quite sure myself what is included in the Starcounter VS extension apart from the templates and the ability to start apps.

**Comment by andwah** [comment]
We are planning to publish the extention on Visual Studio Marketplace https://marketplace.visualstudio.com/

So maybe the links should point to that direction. And on that page we can document the functions of the Extention?

**Comment by per-samuelsson** [comment]
> I'm not quite sure myself what is included in the Starcounter VS extension apart from the templates and the ability to start apps.

That's basically it. To support a simple-as-possible development cycle where you edit-debug-edit-debug, and that works with our somewhat non-conventional model (hosted apps in, currently, a shared process that need to be debugged as standalone applications).

> Although, are there Visual Studio specific things that we might want to document?

Well, it could be worth having some text about how to work with Starcounter applications from within Visual Studio. For example, a section describing that you can pass parameters on the `Properties | Debug | Command line` textbox, and that it support same features as `star.exe` does from there. Those kind of things is kind of what I had in mind the first time around when I was thinking we might want to point users to some Starcounter-In-Visual-Studio-specific landing page.

Another thing that could be worth a few lines is what each template do. We have 2 project templates, and quite a few item templates now, and request for more of those in the pipeline.

> We are planning to publish the extention on Visual Studio Marketplace https://marketplace.visualstudio.com/
> So maybe the links should point to that direction.

Well, I guess we don't want to point users to the marketplace, since they for a fact have the extension installed when those links are clicked (or they are already in the marketplace, clicking from there).

Maybe we could have the "More information" point to some general page on Starcounter, but one that is focused on the developer experience, and that quite early have a section that the common way to create applications is to use our templates.

And in "Getting started", we bring them to a tutorial where the first step is to create a Starcounter application project instance.

**Comment by Mackiovello** [comment]
> Maybe we could have the "More information" point to some general page on Starcounter, but one that is focused on the developer experience, and that quite early have a section that the common way to create applications is to use our templates.

I'll create a page about the VS extension in the "Working with Starcounter" section of the docs that explains the templates and the other page you mentioned. It'll be done before the end of the week.

**Comment by Mackiovello** [comment]
A draft of the page is not created as a PR here: https://github.com/Starcounter/Docs/pull/30

Please take a look and notify if there's something you'd like to add.

**Comment by per-samuelsson** [comment]
Everything here is now done, including updating links. I'm now using the new great page created by @Mackiovello to land "More information", and our tutorial index page for "Getting started".

@mmnosek, just reopen if you find anything should not work to your satisfaction!

**Comment by mmnosek** [comment]
LGTM, thanks a 💯 💯 !
